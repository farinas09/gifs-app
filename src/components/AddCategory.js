import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setinputValue] = useState('');

    const handleInputChange = (e) => {
        setinputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if( inputValue.trim().length > 2){
            //Arrow function has access to current state
        setCategories( cats => [inputValue, ...cats]);
    }
    setinputValue('');
    }

    return (
            <form className="form-group" onSubmit={handleSubmit}>
            <input className="form-control"
                placeholder="Search GIFs"
                type="text"
                value= {inputValue}
                onChange={handleInputChange}
            />
            </form>
    )
}

//required function setCategories
AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired
}
