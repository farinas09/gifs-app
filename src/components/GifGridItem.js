import React from 'react'
import { PropTypes } from 'prop-types'

export const GifGridItem = ({id, title, url}) => {
    console.log(id, title, url)
    return (
        <div className="card animate__animated animate__fadeIn">
            <img className="card-img-top" src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes ={
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
