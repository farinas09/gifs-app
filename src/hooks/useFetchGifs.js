import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";


export const useFetchGifs = (category) => {
    const [state, setstate] = useState({
        data:[],
        loadding: true
    });

    useEffect(() => {
        getGifs(category).then(images => {
            setstate({
                data: images,
                loadding: false
            });
        });
        //empty dependenciesList to call getGifs once
    }, [category]);

    return state;
}