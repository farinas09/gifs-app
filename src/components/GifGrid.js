import React from 'react'
import { PropTypes } from 'prop-types';

import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({category}) => {

    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
        <h3 className="animate__fadeIn">{category}</h3>
        { loading && <div class="alert alert-primary" role="alert">Loading</div> }
        <div className="card-grid album py-5 bg-light animate__fadeIn">
                {
                    images.map( (img) => (
                    <GifGridItem
                        key={img.id}
                        {...img} />
                        )) 
                }
        </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}