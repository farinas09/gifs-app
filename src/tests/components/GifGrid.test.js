
import React from "react"

import '@testing-library/jest-dom';
import { shallow } from "enzyme"
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs")

describe('Pruebas en <GifGrid />', () =>{
    const category = 'category';

    test('should show the component successfully', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        })
        
    const wrapper = shallow(<GifGrid category={category} />)
        expect( wrapper ).toMatchSnapshot();
    })

    test('should show items when usefetch load data', () => {

        const gifs = [{
            id: 'abc',
            url: 'https://xdxd.com/png',
            title: 'text'
        }]
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false
        })
        
    const wrapper = shallow(<GifGrid category={category} />)
        expect( wrapper ).toMatchSnapshot();
        
    })
    
})