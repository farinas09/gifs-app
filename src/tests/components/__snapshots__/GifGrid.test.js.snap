// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> should show items when usefetch load data 1`] = `
<Fragment>
  <h3
    className="animate__fadeIn"
  >
    category
  </h3>
  <div
    className="card-grid album py-5 bg-light animate__fadeIn"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="text"
      url="https://xdxd.com/png"
    />
    <GifGridItem
      id="123"
      key="123"
      title="text"
      url="https://xdxd.com/png"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> should show the component successfully 1`] = `
<Fragment>
  <h3
    className="animate__fadeIn"
  >
    category
  </h3>
  <div
    class="alert alert-primary"
    role="alert"
  >
    Loading
  </div>
  <div
    className="card-grid album py-5 bg-light animate__fadeIn"
  />
</Fragment>
`;
